source/common/network/address_impl.cc

```diff
@@ -78,7 +82,14 @@ InstanceConstSharedPtr peerAddressFromFd(int fd) {
 }
 
 int InstanceBase::flagsFromSocketType(SocketType type) const {
+#if defined(__FreeBSD__)
+  int flags = O_NONBLOCK;
+#elif defined(__APPLE__)
```

Comment:
In various places you condition on both __FreeBSD__ and __APPLE__. Why is this? Are you adding FreeBSD and OS X support? Is the FreeBSD part of the OS X support (but then why special __APPLE__ conditions?)?